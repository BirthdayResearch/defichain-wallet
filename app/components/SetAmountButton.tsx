import React from 'react'
import { TouchableOpacity } from 'react-native'
import { tailwind } from '../tailwind'
import { Text } from './Text'
import { translate } from '../translations'
import BigNumber from 'bignumber.js'

interface SetAmountButtonProps {
  type: 'half' | 'max'
  onPress: (amount: string) => void
  amount: BigNumber
}

export function SetAmountButton (props: SetAmountButtonProps): JSX.Element {
  const decimalPlace = 8

  return (
    <TouchableOpacity
      testID={`button_${props.type}_amount`}
      style={[
        tailwind('flex px-2 py-1.5 border border-gray-300 rounded'),
        props.type === 'half' && tailwind('mr-1')
      ]}
      onPress={() => {
        props.onPress(props.type === 'half' ? props.amount.div(2).toFixed(decimalPlace) : props.amount.toFixed(decimalPlace))
      }}
    >
      <Text style={tailwind('text-primary text-center font-medium')}>{translate('components/max', props.type === 'half' ? '50%' : 'MAX')}</Text>
    </TouchableOpacity>
  )
}
