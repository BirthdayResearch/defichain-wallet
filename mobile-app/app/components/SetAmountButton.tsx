import BigNumber from 'bignumber.js'

import { tailwind } from '@tailwind'
import { translate } from '@translations'
import { ThemedText, ThemedTouchableOpacity } from './themed'

export enum AmountButtonTypes {
  half = '50%',
  max = 'MAX'
}

interface SetAmountButtonProps {
  type: AmountButtonTypes
  onPress: (amount: string) => void
  amount: BigNumber
  customText?: string
}

export function SetAmountButton (props: SetAmountButtonProps): JSX.Element {
  const decimalPlace = 8
  const text = props.customText !== undefined ? props.customText : translate('components/max', props.type)
  return (
    <ThemedTouchableOpacity
      dark={tailwind('border border-gray-400')}
      light={tailwind('border border-gray-300')}
      onPress={() => {
        props.onPress(props.type === AmountButtonTypes.half ? props.amount.div(2).toFixed(decimalPlace) : props.amount.toFixed(decimalPlace))
      }}
      style={[
        tailwind('flex px-2 py-1 rounded'),
        props.type === AmountButtonTypes.half && tailwind('mr-1')
      ]}
      testID={`${props.type}_amount_button`}
    >
      <ThemedText
        dark={tailwind('text-darkprimary-500')}
        light={tailwind('text-primary-500')}
        style={tailwind('text-center font-medium')}
      >
        {text}
      </ThemedText>
    </ThemedTouchableOpacity>
  )
}
