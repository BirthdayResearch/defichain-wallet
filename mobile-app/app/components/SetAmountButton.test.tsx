import { render } from '@testing-library/react-native'
import BigNumber from 'bignumber.js'
import { AmountButtonTypes, SetAmountButton } from './SetAmountButton'

jest.mock('@shared-contexts/ThemeProvider')
const buttonType: AmountButtonTypes[] = [AmountButtonTypes.half, AmountButtonTypes.max]
const buttonAmount = new BigNumber(10)

describe('set amount button', () => {
  buttonType.forEach(type => {
    it(`should match styling of set amount button type ${type}`, () => {
      const onPress = jest.fn()
      const component = (
        <SetAmountButton
          amount={buttonAmount}
          onPress={onPress}
          type={type}
        />
      )
      const rendered = render(component)
      expect(rendered.toJSON()).toMatchSnapshot()
    })
  })
})
