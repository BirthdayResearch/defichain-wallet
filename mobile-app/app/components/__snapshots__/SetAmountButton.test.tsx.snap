// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set amount button should match styling of set amount button type 50% 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "rgba(212, 212, 212, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 1,
      "borderLeftColor": "rgba(212, 212, 212, 1)",
      "borderLeftWidth": 1,
      "borderRightColor": "rgba(212, 212, 212, 1)",
      "borderRightWidth": 1,
      "borderTopColor": "rgba(212, 212, 212, 1)",
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 1,
      "display": "flex",
      "marginRight": 4,
      "opacity": 1,
      "paddingBottom": 4,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 4,
    }
  }
  testID="50%_amount_button"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Array [
          Object {
            "fontFamily": "MediumFont",
            "fontWeight": "500",
            "textAlign": "center",
          },
          Object {
            "color": "rgba(255, 0, 175, 1)",
          },
        ],
      ]
    }
  >
    50%
  </Text>
</View>
`;

exports[`set amount button should match styling of set amount button type MAX 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "rgba(212, 212, 212, 1)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 1,
      "borderLeftColor": "rgba(212, 212, 212, 1)",
      "borderLeftWidth": 1,
      "borderRightColor": "rgba(212, 212, 212, 1)",
      "borderRightWidth": 1,
      "borderTopColor": "rgba(212, 212, 212, 1)",
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 1,
      "display": "flex",
      "opacity": 1,
      "paddingBottom": 4,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 4,
    }
  }
  testID="MAX_amount_button"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Array [
          Object {
            "fontFamily": "MediumFont",
            "fontWeight": "500",
            "textAlign": "center",
          },
          Object {
            "color": "rgba(255, 0, 175, 1)",
          },
        ],
      ]
    }
  >
    MAX
  </Text>
</View>
`;
